Raspbian build fails due to need for libatomic

I'm building CapnP 0.9.1 from source on a new Raspberry Pi 4 running Raspbian (because apt only has version 0.7.) 

```
$ curl -O https://capnproto.org/capnproto-c++-0.9.1.tar.gz
$ tar zxf capnproto-c++-0.9.1.tar.gz
$ cd capnproto-c++-0.9.1
$ ./configure
$ make -j6 check
$ sudo make install
```

Everything went fine until the final "sudo make install" step:

```
...
libtool: link: g++ -I./src -I./src -DKJ_HEADER_WARNINGS -DCAPNP_HEADER_WARNINGS -DCAPNP_INCLUDE_DIR=\"/usr/local/include\" -pthread -O2 -DNDEBUG -pthread -DKJ_HAS_ZLIB -pthread -o .libs/capnp src/capnp/compiler/module-loader.o src/capnp/compiler/capnp.o  ./.libs/libcapnpc.so ./.libs/libcapnp-json.so ./.libs/libcapnp.so ./.libs/libkj.so -lpthread -pthread
/usr/bin/ld: ./.libs/libcapnp.so: undefined reference to `__atomic_load_8'
/usr/bin/ld: ./.libs/libcapnp.so: undefined reference to `__atomic_store_8’
```

After some googling I found that the fix is to link with libatomic -- I simply changed `LIBS` in the Makefile (`LIBS =  -lpthread -latomic`), but I'm aware the Makefile is generated by autoconf; I'm not sure where the list of libraries actually comes from.

--- a/c++/configure.ac
+++ b/c++/configure.ac
@@ -113,6 +113,39 @@ AC_SUBST([CAPNP_LITE_FLAG])
 
 AC_SEARCH_LIBS(sched_yield, rt)
 
+# Check if we need to link against libatomic for atomic operations.
+# On some architectures (e.g., armel, mipsel, powerpc), GCC requires linking
+# with libatomic to support certain atomic operations.
+AC_MSG_CHECKING([if libatomic is needed])
+AC_LINK_IFELSE([
+  AC_LANG_PROGRAM(
+    [[#include <atomic>
+      #include <cstdint>]],
+    [[std::atomic<std::int64_t> a{0};
+      a.fetch_add(1, std::memory_order_relaxed);
+      return 0;]])
+], [
+  AC_MSG_RESULT([no])
+], [
+  # Linking failed, try with -latomic
+  SAVED_LIBS="$LIBS"
+  LIBS="$LIBS -latomic"
+  AC_LINK_IFELSE([
+    AC_LANG_PROGRAM(
+      [[#include <atomic>
+        #include <cstdint>]],
+      [[std::atomic<std::int64_t> a{0};
+        a.fetch_add(1, std::memory_order_relaxed);
+        return 0;]])
+  ], [
+    AC_MSG_RESULT([yes])
+  ], [
+    AC_MSG_RESULT([no, but linking still fails])
+    LIBS="$SAVED_LIBS"
+    AC_MSG_ERROR([Cannot link program using atomic operations. Please ensure your compiler and system support C++ atomics.])
+  ])
+])
+
 # Users will need to use the same -stdlib as us so we'd better let pkg-config know about it.
 STDLIB_FLAG=`echo "$CXX $CXXFLAGS" | grep -o ' [[-]]stdlib=[[^ ]]*'`
 AC_SUBST([STDLIB_FLAG])

